we build lexilla in pre-build, link against it statically
link against scintilla shared lib.

Index: gtk/makefile
--- gtk/makefile.orig
+++ gtk/makefile
@@ -60,13 +60,13 @@ COMPLIB=$(SCINTILLA_DIR)/bin/scintilla.a
 COMPONENT=$(srcdir)/../bin/libscintilla.$(SHAREDEXTENSION)
 LEXILLA=$(srcdir)/../bin/liblexilla.$(SHAREDEXTENSION)
 
-all: $(PROG) $(COMPONENT) $(LEXILLA)
+all: $(PROG)
 
 vpath %.h $(srcdir) $(srcdir)/../src $(LEXILLA_DIR)/include $(SCINTILLA_DIR)/include
 vpath %.cxx $(srcdir) $(srcdir)/../src $(srcdir)/../../lexilla/access $(SCINTILLA_DIR)/call
 
 INCLUDES=-I $(LEXILLA_DIR)/include  -I $(LEXILLA_DIR)/access -I $(SCINTILLA_DIR)/include -I $(srcdir)/../src
-DEFINES += -DGTK
+DEFINES += -DGTK -DLEXILLA_STATIC
 DEFINES += -DPIXMAP_PATH=\"$(pixmapdir)\" -DSYSCONF_PATH=\"$(SYSCONF_PATH)\"
 
 DEFINES += -D$(if $(DEBUG),DEBUG,NDEBUG)
@@ -156,7 +156,7 @@ SRC_OBJS = \
 	Utf8_16.o
 
 $(PROG): SciTEGTK.o Strips.o GUIGTK.o Widget.o DirectorExtension.o $(SRC_OBJS) $(LUA_OBJS)
-	$(CXX) $(BASE_FLAGS) $(LDFLAGS) -rdynamic -Wl,--as-needed -Wl,-rpath,'$${ORIGIN}' -Wl,--version-script $(srcdir)/lua.vers -Wl,-rpath,$(libdir) $^ -o $@ $(CONFIGLIB) $(LIBS) -L ../../scintilla/bin -lscintilla $(LDLIBS)
+	$(CXX) $(BASE_FLAGS) $(LDFLAGS) -rdynamic -Wl,--as-needed -Wl,-rpath,'$${ORIGIN}' -Wl,--version-script $(srcdir)/lua.vers -Wl,-rpath,$(libdir) $^ -o $@ $(CONFIGLIB) $(LIBS) -lscintilla ../../lexilla/bin/liblexilla.a $(LDLIBS)
 
 # Automatically generate header dependencies with "make depend"
 include deps.mak
@@ -166,11 +166,8 @@ include deps.mak
 # Dead:	install -D SciTEGTK.properties $(SYSCONF_PATH)/SciTEGlobal.properties
 install:
 	$(INSTALL) -m 755 -d $(DESTDIR)$(bindir) $(DESTDIR)$(SYSCONF_PATH)
-	$(INSTALL) -m 755 -d $(DESTDIR)$(libdir)
 
 	$(INSTALL) -m 755 $(PROG) $(DESTDIR)$(bindir)
-	$(INSTALL) -m 755 $(COMPONENT) $(DESTDIR)$(libdir)
-	$(INSTALL) -m 755 $(LEXILLA) $(DESTDIR)$(libdir)
 
 	for files in $(srcdir)/../src/*.properties $(srcdir)/../doc/*.html $(srcdir)/../doc/SciTEIco.png $(srcdir)/../doc/PrintHi.png $(srcdir)/../doc/SciTEIndicators.png; \
 	do \
